crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -80,10 +98,10 @@ def activate
   #
   # Provides the restful api call for
   # Deactivate a Repository   /utils/repositories/deactivate   POST   Destroys Repository DataBagItem
-  # required parameters: platform, repo
+  # required parameters: platform, arch, repo
   def deactivate
-    return render_not_found if params[:platform].nil? || params[:repo].nil?
-    ret, _message = ProvisionerService.new(logger).disable_repository(params[:platform], params[:repo])
+    return render_not_found if params[:platform].nil? || params[:arch].nil? || params[:repo].nil?
+    ret, _message = ProvisionerService.new(logger).disable_repository(params[:platform], params[:arch], params[:repo])
```

Comment:
Metrics/LineLength: Line is too long. [118/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)